airflow/providers/exasol/hooks/exasol.py

```diff
@@ -157,15 +161,39 @@ def get_description(statement: ExaStatement) -> Sequence[Sequence]:
             )
         return cols
 
+    @overload
+    def run(
+        self,
+        sql: str | Iterable[str],
+        autocommit: bool = False,
+        parameters: Iterable | Mapping[str, Any] | None = None,
+        handler: None = None,
+        split_statements: bool = False,
+        return_last: bool = True,
+    ) -> None:
+        ...
+
+    @overload
+    def run(
+        self,
+        sql: str | Iterable[str],
+        autocommit: bool = False,
+        parameters: Iterable | Mapping[str, Any] | None = None,
+        handler: Callable[[Any], T] = None,  # type: ignore[assignment]
+        split_statements: bool = False,
+        return_last: bool = True,
+    ) -> T | list[T]:
+        ...
+
     def run(
         self,
         sql: str | Iterable[str],
         autocommit: bool = False,
-        parameters: Iterable | Mapping | None = None,
-        handler: Callable | None = None,
+        parameters: Iterable | Mapping[str, Any] | None = None,
```

Comment:
This is the one package where I'm unsure... 🤔 I'm actually thinking, the old annotation was incorrect and it never allowed for `Iterable` types:

- Note that the original annotation was `Iterable | Mapping | None`. So, Airflow has said for a while that it allows for an iterable.
- `pyexasol` isn't using a SQLAlchemy connection object, as far as I can tell.
- When I look at `pyexasol`'s documentation, and I look at the API https://github.com/exasol/pyexasol/blob/master/docs/REFERENCE.md I don't see any ability to take in a non-mapping iterable type as a valid input. The reason the other connections take in an `Iterable` is because they all wrap SQLAlchemy. But if you look carefully, you'll notice `pyexasol` is completely foregoing SQLAlchemy.

It's possible the safest thing to do here would be to not touch this file's `parameters` type annotations at all. Someone who is knowledgeable on `pyexasol` should confirm whether I am correct that

Reply:
Annotations in providers can be inaccurate since not all of them are very often used. I’d say we can just change the annotation and see if anyone complains.

Reply:
Oh no, mypy yelled at me for violating the Liskov substitution principle when I swap to `Mapping[str, Any] | None`, of course... 🤦 Ahh. I'm just going to keep as-is. I don't want to mess around too much with this. I hope that is OK.